File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashMap = {};
        for i in range(len(nums)):
            hashMap[nums[i]] = i;
        for i in range(len(nums)):
            remainder = target - nums[i]
            if(remainder in hashMap and hashMap[remainder] != i):
                return [i, hashMap[remainder]];

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = {};
    nums.forEach((n, i) => {
        if(map[n])  map[n].push(i);
        else map[n] = [i];
    });
    for(let i = 0; i < nums.length; i += 1) {
        const temp = target - nums[i];
        if(map[temp]) {
            const secondIndex = map[temp][map[temp].length - 1];
            if(i !== secondIndex)    return [i, secondIndex];
        }
    }
};